fault-tolerance/fault-tolerance/src/main/java/io/helidon/faulttolerance/FtSupport.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.faulttolerance;
+
+import io.helidon.common.Functions;
+import io.helidon.common.UncheckedException;
+import io.helidon.service.registry.Interception;
+import io.helidon.service.registry.InterceptionContext;
+
+/**
+ * Abstract types for classes that are (usually) code generated.
+ * <p>
+ * The classes implement the
+ * {@link io.helidon.service.registry.Interception.ElementInterceptor#proceed(io.helidon.service.registry.InterceptionContext, io.helidon.service.registry.Interception.Interceptor.Chain, Object...)}
+ * with the appropriate behavior of the FT handler, and have an abstract method to get the configured instance of the handler.
+ * <p>
+ * When generated, the generated code provides the instance either from {@link io.helidon.service.registry.ServiceRegistry},
+ * or it creates a new instance, with configuration taken from annotations, and maybe augmented by a
+ * {@link io.helidon.common.config.Config}.
+ * <p>
+ * If you want to implement extend one of these classes, you must add a service annotation
+ * (i.e. {@link io.helidon.service.registry.Service.Singleton}), and the instance must be named with the signature of the
+ * element you want to intercept (the element must already support interception, such as when annotated with an
+ * {@link io.helidon.service.registry.Interception.Intercepted} meta-annotation).
+ * <p>
+ * Signature is the fully qualified class name + {@code .} + methodName (or field name) + {@code (}
+ * + list of method/constructor fully qualified parameter types separated by a {@code ,} + {@code )},
+ * such as {@code com.example.MyType.myMethod(java.lang.String,java.lang.String)}.
+ * Fields do not have the part with parameters, constructor follows method format, where name is {@code <init>}.
+ */
+public final class FtSupport {
+    private FtSupport() {
+    }
+
+    private static Exception findCause(UncheckedException e) {
+        Throwable cause = e.getCause();
+
+        if (cause instanceof Exception ex) {
+            return ex;
+        }
+
+        return e;
+    }
+
+    /**
+     * A generated service common interface.
```

Comment:
This seems unused, even in #9915.

Shouldn't this be `interface FtMethod extends Interception.ElementInterceptor` and have all abstract classes implement `FtMethod` ?

In this is the case, maybe we can do `FtMethod` in its own class and a new container class `FtMethods` with all abstract classes.

Reply:
I have removed the class.